app.py

```diff
@@ -114,6 +114,7 @@ def start_code_flow():
                                           scopes, request.args.get("forceConsent", False),
                                           request.args.get("allowConsentOptionDeselection", False),
                                           request.args.get("responseType", "code"),
+                                          _config.get("response_mode", "query"),
```

Comment:
I think it's better to leave the decision about the response mode to the client. You changed the backend so that it will work with both modes, which is good. Now the client could decide which mode to use:

`request.args.get("response_mode", "query"),`

The `response_mode` could be then removed from the settings.